python/cudf/cudf/core/column_accessor.py

```diff
@@ -511,7 +511,7 @@ def _select_by_label_list_like(self, key: Any) -> ColumnAccessor:
     def _select_by_label_grouped(self, key: Any) -> ColumnAccessor:
         result = self._grouped_data[key]
         if isinstance(result, cudf.core.column.ColumnBase):
-            return self.__class__({key: result})
+            return self.__class__({key: result}, multiindex=self.multiindex)
```

Comment:
`ColumnAccessor` has used `self.__class__` since its inception AFAICT. The only time `type(x)` and `x.__class__` are different is if `__class__` is overridden, which isn't the case here I think.

I guess this is just mimicking existing style.

Reply:
For consistency with the rest of the code in this file too, I choose to leave it as `self.__class__`.